### PR TITLE
Supply reference to get current token into file upload component

### PR DESCRIPTION
**Pull request by loneil**
Signed-off-by: Lucas ONeil <lucasoneil@gmail.com>

# Description
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2555
The file upload component needs the current token at the time of API call, not the token at instantiation.
Can pass in a function pointer from the Vue component that we can invoke in the custom component, and the Vue instance has access to the refreshed token from the keycloak library. This only seems to work with functions from `method` if they are supplied as options.

Also dependency updates. (see https://blog.vuejs.org/posts/vue-2-7-naruto.html)

**I'll want to do some testing with this on existing forms so don't promote through to prod after review or merge just yet**

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-293

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-293

File: app/frontend/src/components/designer/FormViewer.vue [hunk 1]
```diff
@@ -152,9 +152,7 @@ export default {
         componentOptions: {
           simplefile: {
             config: Vue.prototype.$config,
-            headers: {
-              Authorization: `Bearer ${this.token}`,
-            },
+            tokenFromChefs: this.getCurrentAuthHeader
```
Comment (jujaga) [inline app/frontend/src/components/designer/FormViewer.vue:155]: Style: Better to name this chefsToken (active vs passive tense)

File: components/src/providers/storage/chefs.ts [hunk 2]
```diff
@@ -3,11 +3,19 @@ import NativePromise from 'native-promise-only';
...
+        Object.keys(options.headers).forEach(k => {
+          const v = options.headers[k];
+          xhr.setRequestHeader(k, v);
+        });
+      }
+
+      // Allow manual setting of any supplied headers above, but need to get the latest
+      // token from the containing app to deal with expiries and override auth
+      if (options.tokenFromChefs) {
+        xhr.setRequestHeader('Authorization', options.tokenFromChefs())
```
Comment (jujaga) [inline components/src/providers/storage/chefs.ts:17]: Style: Better to name this chefsToken (active vs passive tense)

**Review by jujaga — APPROVED** [review]
Would recommend an attribute rename, but otherwise looks great. Tentatively approving; awesome work! 👍

File: app/frontend/src/components/designer/FormViewer.vue (hunk 1 above)
Comment (loneil) [reply app/frontend/src/components/designer/FormViewer.vue]: Good idea

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-293

**Review by TimCsaky — APPROVED** [review]

**Pull request merged**